doc/api/next_api_changes/deprecations/24730-TH.rst

```diff
@@ -0,0 +1,24 @@
+rcParams type
+~~~~~~~~~~~~~
+Relying on ``rcParams`` being a ``dict`` subclass is deprecated.
+
+Nothing will change for regular users because ``rcParams`` will continue to
+be dict-like (technically fulfill the ``MutableMapping`` interface).
+
+However, there are some rare known cases that use
+``dict.__getitem__(rcParams, key)`` and
+``dict.__setitem__(rcParams, key, val)`` for direct data access that
+circumvents any validation logic. While this is technically public, it
+depends on the implementation detail that ``rcParams`` is currently a ``dict``
+subclass.
+
```

Comment:
OK, part of my confusion about this was that I didn't quite understand that this  was calling the `super.__getitem__`.  Suggested rewording for this to be clearer:

```suggestion
The `.RcParams` class does validation checking on calls to `.RcParams.__get_item__` and 
`.RcParams.__set_item__`.  However, there are rare cases where we want to circumvents the 
validation logic and directly access the underlying data values.   Previously, this could be 
accomplished via  a call to the parent methods  ``dict.__getitem__(rcParams, key)`` and ``dict.__setitem__(rcParams, key, val)``. 

```